### PR TITLE
feat(override-plan): Return currency mismatch on plan override

### PR DESCRIPTION
**Pull request by rsempe**

### CODE REVIEW

File: app/services/subscriptions/create_service.rb [hunk 1]
```diff
@@ -58,7 +59,15 @@ def call
 
     private
 
-    attr_reader :customer, :plan, :params, :name, :subscription_at, :billing_time, :external_id, :current_subscription
+    attr_reader :customer,
+      :plan,
```
Comment (jdenquin) [inline app/services/subscriptions/create_service.rb:63]: Layout/ArgumentAlignment: Align the arguments of a method call if they span more than one line. (https://rubystyle.guide#no-double-indent)

File: app/services/subscriptions/create_service.rb [hunk 2]
```diff
@@ -58,7 +59,15 @@ def call
...
+      :params,
```
Comment (jdenquin) [inline app/services/subscriptions/create_service.rb:64]: Layout/ArgumentAlignment: Align the arguments of a method call if they span more than one line. (https://rubystyle.guide#no-double-indent)

File: app/services/subscriptions/create_service.rb [hunk 3]
```diff
@@ -58,7 +59,15 @@ def call
...
+      :name,
```
Comment (jdenquin) [inline app/services/subscriptions/create_service.rb:65]: Layout/ArgumentAlignment: Align the arguments of a method call if they span more than one line. (https://rubystyle.guide#no-double-indent)

File: app/services/subscriptions/create_service.rb [hunk 4]
```diff
@@ -58,7 +59,15 @@ def call
...
+      :subscription_at,
```
Comment (jdenquin) [inline app/services/subscriptions/create_service.rb:66]: Layout/ArgumentAlignment: Align the arguments of a method call if they span more than one line. (https://rubystyle.guide#no-double-indent)

File: app/services/subscriptions/create_service.rb [hunk 5]
```diff
@@ -58,7 +59,15 @@ def call
...
+      :billing_time,
+      :external_id,
```
Comment (jdenquin) [inline app/services/subscriptions/create_service.rb:68]: Layout/ArgumentAlignment: Align the arguments of a method call if they span more than one line. (https://rubystyle.guide#no-double-indent)

File: app/services/subscriptions/create_service.rb [hunk 6]
```diff
@@ -58,7 +59,15 @@ def call
...
+      :current_subscription,
```
Comment (jdenquin) [inline app/services/subscriptions/create_service.rb:69]: Layout/ArgumentAlignment: Align the arguments of a method call if they span more than one line. (https://rubystyle.guide#no-double-indent)

**Review by vincent-pochet — APPROVED** [review]

**Pull request merged**